evm_loader/program/src/query.rs

```diff
@@ -0,0 +1,59 @@
+//! `EVMLoader` query account cache.
+
+use std::collections::BTreeMap;
+use serde::{Deserialize, Serialize};
+use solana_program::pubkey::Pubkey;
+
+/// Represents cache of queries to Solana accounts.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AccountCache {
+    metadata_cache: BTreeMap<Pubkey, Metadata>,
+    data_cache: BTreeMap<DataKey, Vec<u8>>,
```

Comment:
We should have one record in cache for the Solana account. If contract query meta or data, we should place both in the cache.